### PR TITLE
Comet supply acceptance tests

### PR DESCRIPTION
**Pull request by scott-silver**

### CODE REVIEW

File: test/Acceptance/Tests/AcceptanceTests/AcceptanceTests.swift [hunk 1]
```diff
@@ -527,21 +614,51 @@ enum Token: Hashable, Equatable {
...
     }
 }
 
 func toWei(tokenAmount: TokenAmount) -> BigUInt {
-    return BigUInt(tokenAmount.amount * pow(10, Float(tokenAmount.token.decimals)))
+    switch tokenAmount {
+        case let .specifiedAmount(a, t):
+            return BigUInt(a * pow(10, Float(t.decimals)))
+        case .max(_):
+            return BigUInt(1) << 256 - 1
```
Comment (scott-silver) [inline test/Acceptance/Tests/AcceptanceTests/AcceptanceTests.swift:660]: via the Swift repl:

```
> print(BigUInt(1) << 256 - 1)
115792089237316195423570985008687907853269984665640564039457584007913129639935
```

File: test/Acceptance/Tests/AcceptanceTests/AcceptanceTests.swift [hunk 2]
```diff
@@ -527,21 +614,51 @@ enum Token: Hashable, Equatable {
...
+                return t
+        }
+    }
+
+    var amount: Float {
+        switch self {
+            case let .specifiedAmount(a, _):
+                return a
+            case .max(_):
+                fatalError("cannot operate on intermediate amount for TokenAmount.max")
```
Comment (scott-silver) [inline test/Acceptance/Tests/AcceptanceTests/AcceptanceTests.swift:650]: not sure if there is a better way to handle this, but we use add and subtract `amount` values throughout the current process, and adding or subtracting from "max" doesn't really make sense

File: test/Acceptance/Tests/AcceptanceTests/AcceptanceTests.swift [hunk 3]
```diff
@@ -121,6 +121,93 @@ let allTests: [AcceptanceTest] = [
             )
         )
     ),
+    .init(
+        name: "Alice supplies, but does not allow enough for quote pay (testCometSupplyInsufficientFunds)",
```
Comment (scott-silver) [inline test/Acceptance/Tests/AcceptanceTests/AcceptanceTests.swift:125]: this list is not exhaustive, but wanted to get some of these tests added so we can start running them

**Review by coburncoburn — COMMENTED** [review]
cool

**Comment by scott-silver** [comment]
closing in favor of #47

**Pull request closed without merging**